readthedocs/projects/forms.py

```diff
@@ -248,6 +248,14 @@ def __init__(self, *args, **kwargs):
         )
 
         per_project_settings = list(self.Meta.per_project_settings)
+
+        # NOTE: we are deprecating this feature.
+        # However, we will keep it available for projects that already using it.
+        # Old projects not using it already or new projects won't be able to enable.
+        if not self.instance.has_feature(Feature.ALLOW_VERSION_WARNING_BANNER):
```

Comment:
We could also check if project has already this enabled, and only allow it for them instead of creating a migration.

```suggestion
        if not self.instance.show_version_warning and not not self.instance.has_feature(Feature.ALLOW_VERSION_WARNING_BANNER):
```

We also need to discard this field from our API

https://github.com/readthedocs/readthedocs.org/blob/b2197275ea8c60ddcf3765a0f1d881ecbc37b49b/readthedocs/api/v3/serializers.py#L585-L585

Reply:
> We could also check if project has already this enabled, and only allow it for them instead of creating a migration.

If we do this, a project that's using the feature won't be able to re-enable it if they disable it --which is a behavior that I want to avoid.

Reply:
> We also need to discard this field from our API

I would keep it for now. We can delete it when we remove the feature completely for all the projects.